builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -876,3 +878,82 @@ func protocolForValue(v string) string {
 	log.Printf("[WARN] Unable to determine valid protocol: no matching protocols found")
 	return protocol
 }
+
+func deleteLingeringLambdaENIs(conn *ec2.EC2, d *schema.ResourceData) error {
+	filter1 := &ec2.Filter{
+		Name:   aws.String("group-id"),
+		Values: []*string{aws.String(d.Id())},
+	}
+	filter2 := &ec2.Filter{
+		Name:   aws.String("description"),
+		Values: []*string{aws.String("AWS Lambda VPC ENI: *")},
+	}
+	filter3 := &ec2.Filter{
+		Name:   aws.String("requester-id"),
+		Values: []*string{aws.String("*:awslambda_*")},
+	}
+	params := &ec2.DescribeNetworkInterfacesInput{
+		Filters: []*ec2.Filter{filter1, filter2, filter3},
+	}
+	networkInterfaceResp, err := conn.DescribeNetworkInterfaces(params)
+
+	if err != nil {
+		return err
+	}
+
+	v := networkInterfaceResp.NetworkInterfaces
+
+	for _, eni := range v {
+		detachNetworkInterfaceParams := &ec2.DetachNetworkInterfaceInput{
```

Comment:
Probably just rare edge case, but we may find ENIs that are not attached, so I think we should have a condition here

e.g.

```
if eni.Attachment != nil {
  // all the detach logic
}

deleteNetworkInterfaceParams := ...
```

Otherwise ENI that is already detached would cause terraform to crash here (reproducible easily by manually detaching the ENI via AWS Console before running `terraform destroy`).